### DISCUSSION TITLE
New MusE web test

### DISCUSSION THREAD

**Discussion by kybos** [muse-sequencer/muse#1144] [Team Posts]
I created a web for MusE as described in github docu (as a subproject under muse-sequencer).
I put the old front page there as index.html (it's incomplete w/o images/css, just for a test).
The web can be already reached under https://muse-sequencer.github.io 
You all have admin access to the new repo.

**Comment by kybos** [comment]
Now it's almost like the original, with a little update ;-)

**Comment by terminator356** [comment]
Nice! Excellent job as far as I can see. Thanks for this.
(I wonder, how long will it be free services?)
I know relatively little about web page design.
But... I have a book on VRML, that I never used. That counts doesn't it?  ;-)

**Comment by kybos** [comment]
I think it will remain free, changing that would make a lot of people very angry (and github very unpopular).
In fact the functionality of the github Pages is quite restricted, only client-side technologies are supported (html, css, js). No complex webs are possible - like e-shops, CMS etc. that need server-side programming (php or similar) and databases.

**Comment by spamatica** [comment]
Looking good! Eh, or rather, it looks like the same old ;)

I agree, they won't start charging for this exact service, they might try limiting it's usefulness to find a niche for a paid service but it's already quite basic so I'm not sure how..

I suppose we can't run our wiki then. But we need to migrate our wiki content somehow. I saw the Surge synth has a wiki, so I suppose they have a wiki service that we can migrate the content to.

VRML though, that's an acronym I haven't heard in a long time... :D

**Comment by terminator356** [comment]
Good morning (?).
[https://github.com/muse-sequencer/muse/wiki](url)

**Comment by kybos** [comment]
There is a Wiki service here on github (it's even visible in the project header, close to the Code, Issues, Pull requests etc.). So it should be easy to put the relevant content there and set links to it from the web whenever required.

**Comment by terminator356** [comment]
You um... do know who owns github now? He he ;-)

**Comment by kybos** [comment]
Yes, I know, it's also one of the reasons I don't use it for my own projects, the other being that I can have private repos in BitBucket for free ;-).

**Comment by terminator356** [comment]
Gotta checkout. Till tomorrow (or today?) see ya.

**Comment by spamatica** [comment]
I don't think M$ is any worse than any other big tech these days. 
Time will tell but if they start limiting what has already been set in motion I see a lot of developers being very angry, and I don't think they have any wish to do that.
Though I guess all that can change when they release github2 that only supports sourcesafe from 1999 :D oh well...
see ya tomorrow, I mean today!

**Comment by kybos** [comment]
Robert, what do you suggest now? You seem to be the only one with access to the content of the current web.
Copying the whole content to the github pages and wiki as a starting point?
Or just move the original content to some subfolder, so we all can easily access it, and make a clean start from scratch (possibly reusing parts of the existing material)?
You proposed something like the Surge synth web, which is more a mobile device oriented site with its scrolling design, but I guess this is what counts nowadays (on mobile it looks quite cool). 
My own web design experience goes a way back and is quite outdated now (classical desktop web design, written completely in a text editor...). I am sure this is not how it's done now. I assume most people use one of the free web generators? These tools probably also ensure that the result is compatible with all kind of devices.

**Comment by spamatica** [comment]
Den tis 17 dec. 2019 kl 19:50 skrev kybos <notifications@github.com>:

> Robert, what do you suggest now? You seem to be the only one with access
> to the content of the current web.
> Copying the whole content to the github pages and wiki as a starting point?
>
Or just move the original content to some subfolder, so we all can easily
> access it, and make a clean start from scratch (possibly reusing parts of
> the existing material)?
>
I guess, I didn't really think it through. Not sure what you mean with
subfolder, I don't think the pages are physically in the file system of the
web server, I think all the data comes from a database.
I could probably dump the database but that is probably not very usable
without some substantial parsing work..

Copy paste from the most relevant pages? I can get you an account on to the
wiki if that helps in any way.

You proposed something like the Surge synth web, which is more a mobile
> device oriented site with its scrolling design, but I guess this is what
> counts nowadays (on mobile it looks quite cool).
> My own web design experience goes a way back and is quite outdated now
> (classical desktop web design, written completely in a text editor...). I
> am sure this is not how it's done now. I assume most people use one of the
> free web generators? These tools probably also ensure that the result is
> compatible with all kind of devices.
>
I saw there were some templates available, maybe we could use one of those?
I have started a little test but as my web skills are very rusty indeed and
I get side tracked by other priorities so I can't promise anything
happening right away with that.

**Comment by spamatica** [comment]
I put it up on a server, what I was playing with:
http://spamatica.com/museweb2020/

It's based off some tutorial I found and is this far just a random mischmasch of misaligned screenshots and texts.
I do like the basic idea though, having images in the background and sort of parallax scrolling very broshure-like information over... It just needs a lot of work to be presentable. Probably it breaks down on a hidpi screen too.

But I'm not fast, if anyone is faster and has a decent idea I won't complain ;)

**Comment by kybos** [comment]
No problem, just wanted to check with you.
It would be nice if you could give me access where possible and where not just dump the whole available content into some folder here (the content must be in a file system somewhere in the end, even if it's on an apache web server and there are only php and/or mysql files, that does not matter).

**Comment by kybos** [comment]
> I put it up on a server, what I was playing with:
> http://spamatica.com/museweb2020/
> 
> It's based off some tutorial I found and is this far just a random mischmasch of misaligned screenshots and texts.
> I do like the basic idea though, having images in the background and sort of parallax scrolling very broshure-like information over... It just needs a lot of work to be presentable. Probably it breaks down on a hidpi screen too.
> 
> But I'm not fast, if anyone is faster and has a decent idea I won't complain ;)

Looks nice.
I certainly won't be fast either, there will be not much from me before mid of January...

**Comment by terminator356** [comment]
Hey that's really nice! Eye candy!

**Comment by spamatica** [comment]
What do you know, I actually did some more!
Same url: http://spamatica.com/museweb2020/

I'm not a web designer either, or any kind of designer for that matter, so I'm just trying things..
I would appreciate some feedback. Do you think this be workable? Improvement ideas?
Only tested on one browser and one screen size so far..

More text would be nice too although I was thinking that the feature list and such would be relegated to second row, a page in the wiki. The front page would be really high level, glossy.
That is... if you agree (or atleast can live ) with that idea? :)

**Comment by spamatica** [comment]
Background images are photos I picked semi randomly so if they feel wrong I can look for others. Or if you have suggestions of your own, gimme.
The first one is actually of my house ;)

**Comment by terminator356** [comment]
Love the backgrounds. Summer. Winter. And train... _tracks_ - hm, being subliminal are we?  He he.
I think they are relaxing. We could get all glitzy and high-tech but I like them.

(I see you have Mr. Ray '73 there. An old favourite. Can't beat it.)

I like where this is going.
That nice first shot of MusE should be enough alone to make anyone remotely interested **salivate** !
For that reason I'd say try to keep the front page uncluttered.
I've seen bad examples where the front is waaay too long, too much information.

On some sites like this the top 'bar' is hidden once you scroll down and reappears
 once you scroll to the top, or even I think some I have seen reappear as soon as 
 you start scrolling northward.

Spelling and grammar look OK to me, well versed, thanks for that, it could be a lot worse.

**Comment by terminator356** [comment]
On the other hand, people today with phones don't like to 'enter' yet another page with yet another click.
They like to scroll with everything in front of them.
So I suppose strike a good balance.

**Comment by kybos** [comment]
You can see on the first screenshot why I think the "Icons from theme" option should be removed ;-). The first three icons top left are just that:
![image](https://user-images.githubusercontent.com/10009541/71232901-18b37480-22f4-11ea-8f2e-ac9e8e2465a3.png)

**Comment by terminator356** [comment]
We'll just call them... ghost icons ;-)
Sorry forgot to turn that setting default off.
Go ahead, or remove it. No objections came in, yet.

I understand you'll be away from tomorrow was it?
If we don't get a chance to speak before then,
 take care and we'll see you on the flip-side.
Happy 2020.

**Comment by kybos** [comment]
I brushed up my web design knowledge a little, here some points I learned:
1. Mobile devices have the highest priority. The page design has to be very flexible and able to adjust to any screen size (it's called "responsive" now). Failing to do so also negatively influences Google ranking.
2. Bandwidth still matters a lot, because many people use slow/restricted/expensive mobile data connections. Slow/big pages also negatively influence Google ranking.
3. For mobile devices one long (scrolling) page with all the essential information is optimal. For desktop it's not optimal but acceptable.

So as usual, it's a compromise. Or you can create to separate websites for mobile/desktops.

That said, I like the general design of the new web (and the pictures from Sweden, I stayed many times in the stugas there myself ;-) ). I played with it a little a put in some more information from the old site and tried to make it more "compliant". If you want to look, it's available here: https://muse-sequencer.github.io/new/index.html  (far from being finished, or even 100% mobile-friendly).
If you both rather prefer the cool sayings instead of informational content on the main page, I won't of course object. I personally don't like it much, I'm always looking for meaningful information, but I am probably too old and conservative ;-).

**Comment by kybos** [comment]
> I understand you'll be away from tomorrow was it?
> If we don't get a chance to speak before then,
> take care and we'll see you on the flip-side.
> Happy 2020.

Thank you. Yes, we will be leaving tomorrow (I will have email access but no PC).
I also wish Merry Christmas and Happy 2020 to everybody here!

**Comment by spamatica** [comment]
Thanks guys,

Den fre 20 dec. 2019 kl 07:25 skrev kybos <notifications@github.com>:

> I brushed up my web design knowledge a little, here some points I learned:
>
>    1. Mobile devices have the highest priority. The page design has to be
>    very flexible and able to adjust to any screen size (it's called
>    "responsive" now). Failing to do so also negatively influences Google
>    ranking.
>
> Yeah, I know about that part and it is using the bootstrap framework which
is supposed to help scale it for mobile. Though I haven't actually tested
it.


>
>    1. Bandwidth still matters a lot, because many people use
>    slow/restricted/expensive mobile data connections. Slow/big pages also
>    negatively influence Google ranking.
>
>
Indeed, I started shrinking one of the photos but they need to come down in
size all of them.


>    1. For mobile devices one long (scrolling) page with all the essential
>    information is optimal. For desktop it's not optimal but acceptable.
>
> I can agree with that, can't have everything.


>
>
> So as usual, it's a compromise. Or you can create to separate websites for
> mobile/desktops.
>
> That said, I like the general design of the new web (and the pictures from
> Sweden, I stayed many times in the stugas there myself ;-) ). I played with
> it a little a put in some more information from the old site and tried to
> make it more "compliant". If you want to look, it's available here:
> https://muse-sequencer.github.io/new/index.html (far from being finished,
> or even 100% mobile-friendly).
> If you both rather prefer the cool sayings instead of informational
> content on the main page, I won't of course object. I personally don't like
> it much, I'm always looking for meaningful information, but I am probably
> too old and conservative ;-).
>
Haha, not entirely sure what you mean with cool sayings, after rereading
what I had tried to write I felt it was too much salesman.
It needs more work but I still think the feature list should not be placed
on the front page apart from the most obvious ones as it will not look nice.

Didn't check your updated version yet, I'm out of time. Maybe we have a
start that we can work from anyway :)

Regarding age, I'm thinking we are probably the same generation. 49 in a
few weeks..

**Comment by kybos** [comment]
> It needs more work but I still think the feature list should not be placed
> on the front page apart from the most obvious ones as it will not look nice.

Right, the feature list certainly should not be at the top (of the scrolling page), that's the place for a nice colourful eye-catcher, but then, when the user is interested, he should be able to get it with one simple swipe, otherwise he looses interest. People are very impatient nowadays... Links/references do not work so well on (small) touch screens, unless you make them really big.

And yes, we are of the same generation, me only a little bit older...

**Comment by spamatica** [comment]
New try
http://spamatica.com/museweb2020
I took a few ideas from your page kybos and reworked it into my proposal
+ news
+ feature list
+ videos

A lot of minor changes, for instance the header now scrolls off, previously it would stay on top always.
Here's the old one: http://spamatica.com/museweb2020_v1/

Still lots to refine, including that there are multiple fonts used, I think bootstrap tries to use one font whereas I'm trying to test another.

**Comment by spamatica** [comment]
Also I just found out that the releases page on github can actually upload binaries., I don't think it could do that before, or I just didn't understand how... 
We have used sourceforge as file server since forever, might be time to switch...

**Comment by spamatica** [comment]
Hi guys,

Andrew is nowhere to be found... it seems we need to proceed with what we
can do, moving the web to github.
I did some work over the holidays moving more stuff to the github wiki etc.
I would appreciate if anyone could glance through it though, I don't recall
of the top of my head if there are any glaring points missing.

Regarding the new web page proposal, as I wrote before I added more
technical information. At least the bottom of it is unfinished.
https://muse-sequencer.github.io/







Den fre 20 dec. 2019 kl 11:27 skrev kybos <notifications@github.com>:

**Comment by terminator356** [comment]
Looks good to me so far. Spelling mistake:
"The forth and probably last pre-release!"
I didn't know we had a Forth language version, ha ha.

**Comment by danvd** [comment]
Hi Guys,

Sorry for no reply from me for a long time.
Just checked my mailbox and found out that I'm missing :(.
I'm rolling like a squirrel so far. Some time this will have change I hope :).
If I can help with something - let me know. BTW the new site is very nice!
The current site is in qcow2 format - I can share it w/o any problem.

Regards,
Andrew.

**Comment by spamatica** [comment]
Hi Andrew, you're alive :)
Sounds like life is busy though. I have just moved the domain (tried atleast) to github, that seems the best course of action. So you are off the hook for keeping the site running ;)
You're much welcome to come back and help us out with the development though =) 

And Tim, of course there is a Forth version, didn't you hear, that's the next big thing? ;)

**Comment by kybos** [comment]
Thanks Robert for moving the site and the wiki, I think now everything important is there.
I tested a little bit, it looks like the standard domain www.muse-sequencer.org is still not functional, there is always a connection error (both http and https). But the apex domain muse-sequencer.org already points to the new github pages.
I entered the apex domain in the web site repo settings as Custom domain for a test, but in fact github recommends to enter the www subdomain there, for all the mappings to work as supposed to.

**Comment by spamatica** [comment]
Ah, I see. I was just wondering what happened.

You edited the CNAME, right? This does not seem to have worked as intended
it does not show at all now (too many redirections)
What exactly was it you were going to fix?

I neglected to add www but I think this is at the domain provider.
I'll add it at the provider and take it from there?


Den fre 17 jan. 2020 kl 20:14 skrev kybos <notifications@github.com>:

**Comment by kybos** [comment]
In fact I did not edit CNAME, I only entered the domain here:
https://github.com/muse-sequencer/muse-sequencer.github.io/settings
(below in Git Pages->Custom domain)
This probably overwrote it.
Sorry if I messed up something...

**Comment by spamatica** [comment]
Den fre 17 jan. 2020 kl 20:59 skrev kybos <notifications@github.com>:

> In fact I did not edit CNAME, I only entered the domain here:
> https://github.com/muse-sequencer/muse-sequencer.github.io/settings
> (below in Git Pages->Custom domain)
> This probably overwrote it.
> Sorry if I messed up something...
>
Yeah, that writes the CNAME file which in turn causes it to forward to
muse-sequencer.org (which in turn will forward it back), I disabled this
again.

I could not get it to work using the muse-sequencer.org domain due to
certificate issues so I settled on forwarding to the github.io address. I
guess it could have been a temporary issue and would have solved itself
over time..
Though I guess I saw it as a way to some day move away from the domain that
cost money also.

Regarding www, I didn't get it to work yet, there is a edit field which
seems like it is used for this at the domain provider but I'm not allowed
to edit it..

**Comment by kybos** [comment]
@danvd 
> The current site is in qcow2 format - I can share it w/o any problem.

Yes please, make the content of the "old" site available if possible.
I remember there were some more wiki pages that are not yet copied to the new web (some of them quite hidden). Most of the content is certainly obsolete, but there was also artwork etc. that could still be useful.

**Comment by kybos** [comment]
> Though I guess I saw it as a way to some day move away from the domain that
> cost money also.

Sure, that's also an aspect to be considered.

Just my personal opinion:
The new github.io domain is (almost ;-) ) just as good as the "old" one, I don't see any reason why you should pay for it. People (and Google) would get used to it soon... 
I would even say that just now - given the recent chaos with the web site access - would be the perfect point in time to switch to the new github domain completely. 
In that case, if you still have the old domain subscribed for some time, you could e.g. put a simple html page at the old address with the info that the site has been moved and a link to the new location.

Other opinions on that?

**Comment by kybos** [comment]
I updated the web, I started mainly because it did not really work on mobile (at least on my IPhone X, tried both Safari and Firefox). More details in the PR:
https://github.com/muse-sequencer/muse-sequencer.github.io/pull/1
If you don't like the colors or have a better screenshot etc. it can of course be changed. But now it should generally work on all kind of devices.
There is still an issue with the background images on my phone, I will look into it.

**Comment by spamatica** [comment]
As I wrote on the PR
When first opening it I wasn't sure about the color changes (the stronger
background colors) but it's fine, good work! And the images are clickable
now.
The new arranger view is a composition of yours?

I didn't notice any big issues when browsing the page on my phone earlier
but then I don't have an iphone.
I believe all browsers on iphone are forced to safaris rendering engine so
it might be safari specific. Tried it now with the fixes, no issues that I
can see.

Ah, tiny issue with the News link, it scrolls a bit past the beginning. No
idea why that would be.


Den mån 20 jan. 2020 kl 12:23 skrev kybos <notifications@github.com>:

**Comment by kybos** [comment]
> The new arranger view is a composition of yours?

No, a midi file from the web (I think it was Genesis), and bounced to track so there is some audio too...
 
> I didn't notice any big issues when browsing the page on my phone earlier
> but then I don't have an iphone.
> I believe all browsers on iphone are forced to safaris rendering engine so
> it might be safari specific. Tried it now with the fixes, no issues that I
> can see.

Yes. The Safari engine seems to be quite buggy, or not quite up-to-date with modern technologies...

> Ah, tiny issue with the News link, it scrolls a bit past the beginning. No
> idea why that would be.

I noticed that too. It's because the anchor points to the top of the page, and there is the navbar. I will see what I can do.

And thanks for checking.

**Comment by kybos** [comment]
> Ah, tiny issue with the News link, it scrolls a bit past the beginning. No
> idea why that would be.

That should be fixed now.

I learned that fixed backgrounds are not supported at all in Safari mobile - a deliberate decision by Apple, they claim they are too heavy on the CPU. Stupidly enough, they display a blurry something instead of just scrolling the background image (like it would without the "fixed" attribute). A special CSS hack is necessary to make that work...

**Comment by kybos** [comment]
The screenshots wiki page (https://github.com/muse-sequencer/muse/wiki/Screenshots) contains only broken links. Do we have the corresponding pictures somewhere?

**Comment by spamatica** [comment]
Den tis 21 jan. 2020 kl 07:49 skrev kybos <notifications@github.com>:

> The screenshots wiki page (
> https://github.com/muse-sequencer/muse/wiki/Screenshots) contains only
> broken links. Do we have the corresponding pictures somewhere?
>

Right, I think I made backups of almost everything but I don't have it in
reach now. Will try to put it somewhere tonight.
Hopefully we can get the hands of the vm image file as a complete backup, I
should have most of it though.

/Robert

**Comment by spamatica** [comment]
Checked last night and I had nearly completely failed with my backup
attempt...
I have most of the meta data but no images...

I recalled wayback machine though and it seems it has most of it.
E.g.
https://web.archive.org/web/20190707011109/http://muse-sequencer.org/index.php/Screenshots

Den tis 21 jan. 2020 kl 11:05 skrev Robert Jonsson <spamatica@gmail.com>:

>
>
> Den tis 21 jan. 2020 kl 07:49 skrev kybos <notifications@github.com>:
>
>> The screenshots wiki page (
>> https://github.com/muse-sequencer/muse/wiki/Screenshots) contains only
>> broken links. Do we have the corresponding pictures somewhere?
>>
>
> Right, I think I made backups of almost everything but I don't have it in
> reach now. Will try to put it somewhere tonight.
> Hopefully we can get the hands of the vm image file as a complete backup,
> I should have most of it though.
>
> /Robert
>

**Comment by kybos** [comment]
OK, I restored the screenshot wiki page to the (almost) original state. An interesting version history, BTW...
The next thing would be to add some more recent screenshots, as the last one is from 2.1...

**Comment by spamatica** [comment]
Right, I have a few, just need to get it done..

Den lör 25 jan. 2020 kl 20:37 skrev kybos <notifications@github.com>:
